### PR TITLE
pgsrip: init at 0.1.11

### PR DESCRIPTION
**Pull request by eljamm**
## Description of changes

- pgsrip: tool to rip PGS subtitles
- trakit: guess additional information from track titles (dependency)
- cleanit: command-line tool to clean subtitles (dependency)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by eljamm** [comment]
Result of `nixpkgs-review pr 316209` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>cleanit (python311Packages.cleanit)</li>
    <li>cleanit.dist (python311Packages.cleanit.dist)</li>
    <li>pgsrip (python311Packages.pgsrip)</li>
    <li>pgsrip.dist (python311Packages.pgsrip.dist)</li>
    <li>trakit (python311Packages.trakit)</li>
    <li>trakit.dist (python311Packages.trakit.dist)</li>
    <li>python312Packages.trakit</li>
    <li>python312Packages.trakit.dist</li>
  </ul>
</details>

**Comment by eljamm** [comment]
Result of `nixpkgs-review pr 316209` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>cleanit (python312Packages.cleanit)</li>
    <li>cleanit.dist (python312Packages.cleanit.dist)</li>
    <li>pgsrip (python312Packages.pgsrip)</li>
    <li>pgsrip.dist (python312Packages.pgsrip.dist)</li>
    <li>python311Packages.cleanit</li>
    <li>python311Packages.cleanit.dist</li>
    <li>python311Packages.pgsrip</li>
    <li>python311Packages.pgsrip.dist</li>
    <li>python311Packages.trakit</li>
    <li>python311Packages.trakit.dist</li>
    <li>trakit (python312Packages.trakit)</li>
    <li>trakit.dist (python312Packages.trakit.dist)</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:1]: Do all of those new packages have scripts which are often used on their own? Why do they then not have meta.mainProgram?

File: pkgs/development/python-modules/pgsrip/default.nix [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  nativeCheckInputs = [
+    flake8
+    mypy
+  ];
+
+  checkPhase = ''
+    flake8
+    mypy --check-untyped-defs pgsrip
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pgsrip/default.nix:67]:
```suggestion
```
we don't care about lints

File: pkgs/development/python-modules/pgsrip/default.nix [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+  patchPhase = ''
+    substituteInPlace pyproject.toml --replace "opencv-python" "opencv"
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pgsrip/default.nix:71]:
```suggestion
  postPatch = ''
    substituteInPlace pyproject.toml \
      --replace-fail "opencv-python" "opencv"
  '';
```

**Comment by eljamm** [comment]
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>cleanit (python312Packages.cleanit)</li>
    <li>cleanit.dist (python312Packages.cleanit.dist)</li>
    <li>pgsrip (python312Packages.pgsrip)</li>
    <li>pgsrip.dist (python312Packages.pgsrip.dist)</li>
    <li>python311Packages.cleanit</li>
    <li>python311Packages.cleanit.dist</li>
    <li>python311Packages.pgsrip</li>
    <li>python311Packages.pgsrip.dist</li>
    <li>python311Packages.trakit</li>
    <li>python311Packages.trakit.dist</li>
    <li>python312Packages.trakit</li>
    <li>python312Packages.trakit.dist</li>
  </ul>
</details>

**Comment by eljamm** [comment]
Moved `pgsrip` to be an application since we don't really need its module.

---

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>cleanit (python312Packages.cleanit)</li>
    <li>cleanit.dist (python312Packages.cleanit.dist)</li>
    <li>pgsrip</li>
    <li>pgsrip.dist</li>
    <li>python311Packages.cleanit</li>
    <li>python311Packages.cleanit.dist</li>
    <li>python311Packages.trakit</li>
    <li>python311Packages.trakit.dist</li>
    <li>python312Packages.trakit</li>
    <li>python312Packages.trakit.dist</li>
  </ul>
</details>

**Comment by eljamm** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 316209`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>cleanit (python312Packages.cleanit)</li>
    <li>cleanit.dist (python312Packages.cleanit.dist)</li>
    <li>pgsrip</li>
    <li>pgsrip.dist</li>
    <li>python311Packages.cleanit</li>
    <li>python311Packages.cleanit.dist</li>
    <li>python311Packages.trakit</li>
    <li>python311Packages.trakit.dist</li>
    <li>python312Packages.trakit</li>
    <li>python312Packages.trakit.dist</li>
  </ul>
</details>

**Comment by eljamm** [comment]
@ofborg build pgsrip

File: pkgs/by-name/pg/pgsrip/package.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  fetchFromGitHub,
+  nix-update-script,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "pgsrip";
+  version = "0.1.11";
+  pyproject = true;
+
+  disabled = python3Packages.pythonOlder "3.9";
```
Comment (SuperSandro2000) [inline pkgs/by-name/pg/pgsrip/package.nix:13]: This line is a bit useless because python3Packages will never be 3.9 or below.

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/by-name/pg/pgsrip/package.nix (hunk 3 above)
Comment (eljamm) [reply pkgs/by-name/pg/pgsrip/package.nix:13]: Maybe not by default, but I don't think that it hurts to include this information in any case.

**Pull request merged**